### ISSUE TITLE
astropy v7.0 incompatibility in TestMasksTracing.test_fit_trace_all_nan_cols

### ISSUE THREAD

**Issue by pllim** [astropy/specreduce#229]
According to [astropy-integration-testing run](https://github.com/astropy/astropy-integration-testing/actions/runs/11785052451/job/32825493170), there is failure with both stable and dev versions, so this package needs a compatibility fix and a new release to be compatible with `astropy` v7.0.0. FYI.

**Comment by pllim** [comment]
And looks like you been warned of this for a long time in your cron jobs but it was left unhandled.

**Comment by pllim** [comment]
```
>       np.testing.assert_allclose(truth, max_trace.trace)
E       AssertionError: 
E       Not equal to tolerance rtol=1e-07, atol=0
E       
E       Mismatched elements: 11 / 11 (100%)
E       Max absolute difference among violations: 0.00700257
E       Max relative difference among violations: 0.00360873
E        ACTUAL: array([1.947455, 2.383634, 2.819813, 3.255992, 3.692171, 4.12835 ,
E              4.564529, 5.000708, 5.436887, 5.873067, 6.309246])
E        DESIRED: MaskedArray([1.940452, 2.376862, 2.813273, 3.249683, 3.686093, 4.122503,
E                    4.558913, 4.995323, 5.431733, 5.868143, 6.304553])
```

This test already ignores a bunch of warnings, so maybe not a well designed test to begin with. Just increase the tolerance?

Also appears to be using a fitter that is no longer recommended, see:

* https://github.com/astropy/astropy/pull/16983

**Comment by hpparvi** [comment]
This should be fixed now in PR #232.
